hxckr-org/hxckr-core
# feat: add migrations for local dev and production

# Separate Development and Production Docker Builds

## Changes
- Created a new `Dockerfile.dev` for development environments
- Added an `entrypoint.dev.sh` script to run migrations in development
- Updated GitHub Actions workflow to build separate development and production images
- Moved production migration step to a separate action in the workflow

## Why
This change separates our development and production Docker builds, allowing us to:
1. Run migrations automatically in development environments
2. Keep our production image lean and flexible
3. Run production migrations as a separate step for better control and visibility

## How to Test
1. Build the development image locally:
   ```
   docker build -f Dockerfile.dev -t hxckr-core:dev .
   ```
2. Run the development container:
   ```
   docker run -e DATABASE_URL=your_local_db_url hxckr-core:dev
   ```
3. Verify that migrations run and the application starts

## Production Deployment Steps
1. Create a new release in GitHub
2. The workflow will automatically build and push the production image
3. The workflow will run migrations using the production image

## Setup Required
To enable production deployments, you need to set up the following secrets in your GitHub repository:

1. `PRODUCTION_DATABASE_URL`: The URL of your production database

To set up these secrets:
1. Go to your GitHub repository
2. Click on "Settings" > "Secrets and variables" > "Actions"
3. Click "New repository secret"
4. Add each secret with its respective name and value

## Future Improvements
- Implement database rollback capabilities for failed migrations
- Add more comprehensive testing in the CI pipeline before building production images
